src/Data/Bin/Ordering.agda

```diff
@@ -0,0 +1,666 @@
+
+toℕ-mono-< :  toℕ Preserves _<_ ⟶ ℕ._<_
+toℕ-mono-< {_}        {zero}     ()
+toℕ-mono-< {zero}     {2[1+ _ ]} _               =  ℕₚ.0<1+n
+toℕ-mono-< {zero}     {1+[2 _ ]} _               =  ℕₚ.0<1+n
+toℕ-mono-< {2[1+ x ]} {2[1+ y ]} (even<even x<y) =  begin
+  ℕ.suc (2 ℕ.* (ℕ.suc xN))    ≤⟨ ℕₚ.+-monoʳ-≤ 1 (ℕₚ.*-monoʳ-≤ 2 xN<yN) ⟩
+  ℕ.suc (2 ℕ.* yN)            ≤⟨ ℕₚ.≤-step ℕₚ.≤-refl ⟩
+  2 ℕ.+ (2 ℕ.* yN)            ≡⟨ sym (ℕₚ.*-distribˡ-+ 2 1 yN) ⟩
+  2 ℕ.* (ℕ.suc yN)            ∎
+  where open ℕₚ.≤-Reasoning;  xN = toℕ x;  yN = toℕ y;  xN<yN = toℕ-mono-< x<y
+
+toℕ-mono-< {2[1+ x ]} {1+[2 y ]} (even<odd x<y) =  ℕₚ.+-monoʳ-≤ 1 (ℕₚ.*-monoʳ-≤ 2 xN<yN)
+  where
+  xN = toℕ x;  yN = toℕ y;  xN<yN = toℕ-mono-< x<y
+
+toℕ-mono-< {1+[2 x ]} {2[1+ y ]} (odd<even (inj₁ x<y)) =   begin
+  ℕ.suc (ℕ.suc (2 ℕ.* xN))    ≡⟨ refl ⟩
+  2 ℕ.+ (2 ℕ.* xN)            ≡⟨ sym (ℕₚ.*-distribˡ-+ 2 1 xN) ⟩
+  2 ℕ.* (ℕ.suc xN)            ≤⟨ ℕₚ.*-monoʳ-≤ 2 xN<yN ⟩
+  2 ℕ.* yN                    ≤⟨ ℕₚ.*-monoʳ-≤ 2 (ℕₚ.≤-step ℕₚ.≤-refl) ⟩
+  2 ℕ.* (ℕ.suc yN)            ∎
+  where open ℕₚ.≤-Reasoning;  xN = toℕ x;  yN = toℕ y;  xN<yN = toℕ-mono-< x<y
+
+toℕ-mono-< {1+[2 x ]} {2[1+ .x ]} (odd<even (inj₂ refl)) =
+  ℕₚ.≤-reflexive (sym (ℕₚ.*-distribˡ-+ 2 1 xN))  where xN = toℕ x
+
+toℕ-mono-< {1+[2 x ]} {1+[2 y ]} (odd<odd x<y) =  ℕₚ.+-monoʳ-< 1 (ℕₚ.*-monoʳ-< 1 xN<yN)
+  where xN = toℕ x;  yN = toℕ y;  xN<yN = toℕ-mono-< x<y
+
+fromℕ-mono-< :  fromℕ Preserves ℕ._<_ ⟶ _<_
+fromℕ-mono-< {m} {n} m<n =  aux (<-cmp x y)
+  where
+  open P.≡-Reasoning
+  x       = fromℕ m;      y       = fromℕ n
+  toℕ-x≡m = toℕ-fromℕ m;  toℕ-y≡n = toℕ-fromℕ n
+
+  aux : Tri (x < y) (x ≡ y) (x > y) → x < y
+  aux (tri< x<y _   _) =  x<y
+  aux (tri≈ _   x≡y _) =  contradiction m≡n (ℕₚ.<⇒≢ m<n)
+    where
+    m≡n = begin
+      m       ≡⟨ sym toℕ-x≡m ⟩
+      toℕ x   ≡⟨ cong toℕ x≡y ⟩
+      toℕ y   ≡⟨ toℕ-y≡n ⟩
+      n       ∎
+  aux (tri> _ _ y<x) =  contradiction m<n (ℕₚ.<⇒≯ n<m)
+    where
+    yN<xN = toℕ-mono-< y<x;  n<m = subst₂ ℕ._<_ toℕ-y≡n toℕ-x≡m yN<xN
+
+
+fromℕ-mono-≤ :  fromℕ Preserves ℕ._≤_ ⟶ _≤_
+fromℕ-mono-≤ m≤n  with ℕₚ.m≤n⇒m<n∨m≡n m≤n
+... | inj₁ m<n =  inj₁ (fromℕ-mono-< m<n)
+... | inj₂ m≡n =  inj₂ (cong fromℕ m≡n)
+
+toℕ-mono-≤ :  toℕ Preserves _≤_ ⟶ ℕ._≤_
+toℕ-mono-≤ (inj₁ x<y)  =  ℕₚ.<⇒≤ (toℕ-mono-< x<y)
+toℕ-mono-≤ (inj₂ refl) =  ℕₚ.≤-reflexive refl
+
+<⇒<ₙon :  ∀ {x y} → x < y → x <ₙon y
+<⇒<ₙon x<y =  toℕ-mono-< x<y
+
+<ₙon⇒< :  ∀ {x y} → x <ₙon y → x < y
+<ₙon⇒< {x} {y} xN<yN =  subst₂ _<_ (fromℕ-toℕ x) (fromℕ-toℕ y) xN'<yN'
+  where
+  xN'<yN' = fromℕ-mono-< xN<yN
+
+------------------------------------------------------------------------------
+-- Reflexivity, transitivity, antisymmetry
+------------------------------------------------------------------------------
+
+≤-refl :  Reflexive _≤_
+≤-refl =  inj₂ refl
+
+≤-reflexive : _≡_ ⇒ _≤_
+≤-reflexive {x} {_} refl =  ≤-refl {x}
+
+≤-trans : Transitive _≤_
```

Comment:
This proof and several others can be found in `Relation.Binary.Construct.StrictToNonStrict`.

Reply:
I set now
```
import Relation.Binary.Construct.StrictToNonStrict _≡_ _<_ as ToNonstrict
...
≤-trans = ToNonstrict.trans isEquivalence (resp₂ _<_) <-trans
...
≤-antisym = ToNonstrict.antisym isEquivalence <-trans <-irrefl
```
Other possible applications of ToNonstrict are not done because I have an
impression that they will not simplify the code.

> Thanks for removing the implicits, I would still kind of prefer that ``_<_`` was defined in 
> terms  of ``_≤_`` but upon reflection, it's probably not a hill that I want to die on so you'll 
> be happy to hear that the current definition can stay

Thank you very much!